### PR TITLE
Add Player Join Event logic

### PR DESCRIPTION
**Pull request by ajarrett29**
@ShepherdJerred did I miss anything?

# Changes Made
added player join event
# Testing Done

### CODE REVIEW

**Comment by ajarrett29** [comment]
I am fixing conflicts

File: src/main/java/com/shepherdjerred/capstone/server/events/events/PlayerJoinEvent.java [hunk 1]
```diff
@@ -3,14 +3,17 @@
...
 import lombok.Getter;
 import lombok.ToString;
 
 @Getter
 @ToString
+@EqualsAndHashCode
 @AllArgsConstructor
 public class PlayerJoinEvent implements Event {
-
+  private final ClientId clientId;
```
Comment (shepherdjerred) [inline src/main/java/com/shepherdjerred/capstone/server/events/events/PlayerJoinEvent.java:17]: A ClientId probably doesn't belong here, but it's not worth the work to find a better design.

File: src/main/java/com/shepherdjerred/capstone/server/events/handlers/PlayerJoinEventHandler.java [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+@AllArgsConstructor
 public class PlayerJoinEventHandler implements EventHandler<PlayerJoinEvent> {
+  private final GameServer gameServer;
+  private final ConnectorHub connectorHub;
 
   @Override
-  public void handle(PlayerJoinEvent event) {
+  public void handle(PlayerJoinEvent playerJoinEvent) {
+    gameServer.addPlayer(playerJoinEvent.getClientId(), playerJoinEvent.getPlayer());
+    connectorHub.sendPacket(new PlayerDescriptionPacket(playerJoinEvent.getPlayer().getName()) {
```
Comment (shepherdjerred) [inline src/main/java/com/shepherdjerred/capstone/server/events/handlers/PlayerJoinEventHandler.java]: Don't see a player description packet, create a new packet type called PlayerJoined (or something similar). Our clients need the entire player object, not just the name.

**Review by shepherdjerred — CHANGES_REQUESTED** [review]

**Comment by shepherdjerred** [comment]
Just fix the PlayerJoinEventHandler and then we can merge it in. It looks good!

**Comment by ajarrett29** [comment]
@ShepherdJerred I made the changes. Take a look.

File: src/main/java/com/shepherdjerred/capstone/server/events/handlers/PlayerJoinEventHandler.java [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+@AllArgsConstructor
 public class PlayerJoinEventHandler implements EventHandler<PlayerJoinEvent> {
+  private final GameServer gameServer;
+  private final ConnectorHub connectorHub;
 
   @Override
-  public void handle(PlayerJoinEvent event) {
+  public void handle(PlayerJoinEvent playerJoinEvent) {
+    gameServer.addPlayer(playerJoinEvent.getClientId(), playerJoinEvent.getPlayer());
+    connectorHub.sendPacket(new PlayerJoinedPacket(playerJoinEvent.getPlayer()) {
```
Comment (shepherdjerred) [inline src/main/java/com/shepherdjerred/capstone/server/events/handlers/PlayerJoinEventHandler.java:19]: Perfect!

**Review by shepherdjerred — CHANGES_REQUESTED** [review]

**Comment by shepherdjerred** [comment]
Sorry I didn't put this in the last review, but there's one more problem. The client won't know what element to send (because they won't know what elements are taken when joining a lobby). For now just leave it hardcoded.

**Comment by shepherdjerred** [comment]
Looks good!

**Pull request merged**